From oneukum@suse.de  Fri Mar  2 13:26:55 2007
From: Oliver Neukum <oneukum@suse.de>
Date: Mon, 26 Feb 2007 15:43:00 +0100
Subject: USB: fix usb-serial device naming bug
To: Craig Schlenter <craig@codefountain.com>
Cc: gregkh@suse.de, davej@redhat.com
Message-ID: <200702261543.00896.oneukum@suse.de>
Content-Disposition: inline


Am Montag, 26. Februar 2007 15:16 schrieb Craig Schlenter:
> Hi Greg
> 
> 34ef50e5b1f96c2d8c0f3d28b7d407743806256c is definitely
> the source of the problem. Reverting that makes the
> ftdi port show up as ttyUSB0 again for me and it
> can actually be opened.

This patch should fix the issue.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Craig Schlenter <craig@codefountain.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/usb-serial.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/usb-serial.c
+++ b/drivers/usb/serial/usb-serial.c
@@ -827,7 +827,6 @@ int usb_serial_probe(struct usb_interfac
 			num_ports = type->num_ports;
 	}
 
-	serial->minor = minor;
 	serial->num_ports = num_ports;
 	serial->num_bulk_in = num_bulk_in;
 	serial->num_bulk_out = num_bulk_out;
@@ -981,6 +980,7 @@ int usb_serial_probe(struct usb_interfac
 		dev_err(&interface->dev, "No more free serial devices\n");
 		goto probe_error;
 	}
+	serial->minor = minor;
 
 	/* register all of the individual ports with the driver core */
 	for (i = 0; i < num_ports; ++i) {
